src/cli.rs

```diff
@@ -36,23 +36,31 @@ fn calculate_complexity(flags: flags::Flags) {
         .for_each(|i| files_filter.only_paths.push(i));
 
     builder.filter_entry(move |e| files_filter.matches(e.path()));
+    builder.threads(num_cpus::get());
 
-    let results = Arc::new(Mutex::new(vec![]));
-    let scorer = flags.scorer;
+    let mut results = vec![];
+    let scorer = &flags.scorer;
+
+    let (sender, receiver) = unbounded();
 
     builder.build_parallel().run(|| {
-        Box::new(|result| {
+        let sender = sender.clone();
```

Comment:
The use of `.clone()` and `drop()` took me by surprise, but I think I get the pun now:

- we clone inside `run` so the Fn can acquire access to the sender
- we drop outside `run` so we close the sender on the main thread

Should we also do the inverse -- drop receiver inside the Fn, clone it on main?

Reply:
I'm not sure I'm understanding your proposal for the inverse portion.

We drop sender after providing the closure to handle entries so the program can proceed (it needs to be dropped prior to the `receiver..recv()`), and then process the entries. Cloning the sender is the MP (multiple producer) portion, and *not* cloning the receiver is the SC (single consumer) portion. I think if there were multiple consumers, it'd push back to having to use a mutex to manage the shared `results` value, which we want to avoid.

Reply:
Oh, I see. Thanks for the MP and SC explanation.